OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -453,6 +453,12 @@ private void UpdateStatus(ShortApplicationDto applicationDto, ApplicationDto app
             }
         }
 
+        if (application.Workshop.CompetitiveSelection && application.Status == ApplicationStatus.Pending &&
+            applicationDto.Status != ApplicationStatus.AcceptedForSelection)
```

Comment:
What if we want to reject a pending application and our workshop has competitive selection? It looks like a valid action but it will cause an exception.
On the other hand there is still missing validation for workshops without competitive selection. It is still possible for them to set `ApplicationStatus` to `AcceptedForSelection`.

Reply:
> What if we want to reject a pending application and our workshop has competitive selection? It looks like a valid action but it will cause an exception.

I thought about this case and discussed with Mykita Babenko.  Provider can reject or accept the application after AcceptedForSelection status only.

We can discuss this case again with Mykita
Imho provider should be able to reject the application without competitive selection.

Reply:
> On the other hand there is still missing validation for workshops without competitive selection. It is still possible for them to set `ApplicationStatus` to `AcceptedForSelection`.

Added validation and testcase for workshops without competitive selection.